baynet/structure.py

```diff
@@ -298,7 +298,8 @@ def estimate_parameters(
     ) -> None:
         """Estimate conditional probabilities based on supplied data."""
         try:
-            self.vs["levels"]
+            if all(vertex["levels"] is None for vertex in self.vs):
```

Comment:
```suggestion
            if any(len(vertex["levels"]) == 0  for vertex in self.vs):
```

If any levels are an empty list, then infer them

Reply:
I don't think this should ever happen though, there must be some other issue

Reply:
> Sometimes DAG.vs['levels'] is List[None] instead of being unset, this catches that eventuality.

Sometimes DAG.vs['levels'] is List[List[Empty]]

Reply:
> I don't think this should ever happen though, there must be some other issue

It definitely does, I promise

Reply:
Oh I don't doubt that, I just mean there's some other underlying issue we've not found - for now I'll just make `infer_levels=True` take priority over existing levels though